frontend/src/components/Candidate-Decider/CandidateDecider.tsx

```diff
@@ -113,13 +151,43 @@ const CandidateDecider: React.FC<CandidateDeciderProps> = ({ uuid }) => {
     <div></div>
   ) : (
     <div className={styles.candidateDeciderContainer}>
+      <Modal open={isOpen} onClose={() => setIsOpen(false)} size="small">
+        <Modal.Header>Don't Forget To Save!</Modal.Header>
+        <Modal.Content>
+          <p>You have unsaved changes. Do you want to save them before navigating?</p>
+        </Modal.Content>
+        <Modal.Actions>
+          <Button onClick={() => setIsOpen(false)}>Cancel</Button>
+          <Button
+            onClick={() => {
+              handleRatingAndCommentChange(
+                currentCandidate,
+                currentRating ?? 0,
+                currentComment ?? ''
+              );
+              confirmNavigation(navigationDirection!);
+            }}
+          >
+            Save and {navigationDirection === 'previous' ? 'Go Back' : 'Proceed'}
+          </Button>
+          <Button primary onClick={() => confirmNavigation(navigationDirection!)}>
+            Discard and {navigationDirection === 'previous' ? 'Go Back' : 'Proceed'}
+          </Button>
+        </Modal.Actions>
+      </Modal>
       <div className={styles.applicationContainer}>
         <div className={styles.searchBar}>
           <SearchBar
             instance={instance}
             setCurrentCandidate={(candidate) => {
-              setCurrentCandidate(candidate);
-              populateReviewForCandidate(candidate);
+              if (!isSaved) {
```

Comment:
can you abstract this function and the below into a helper function? for cleanliness ideally we can put all the handling for this into a single function, as opposed to spreading the logic around.